charms/kserve-controller/src/charm.py

```diff
@@ -76,6 +97,31 @@ def _context(self):
             "cert": f"'{ca_context.decode('utf-8')}'",
         }
 
+    @property
+    def _inference_service_context(self):
+        """Context for rendering the inferenceservive-config ConfigMap."""
+        # Ensure any input is valid for deployment mode
+        deployment_mode = self.model.config["deployment-mode"].lower()
+        if deployment_mode == "serverless":
+            deployment_mode = "Serverless"
+        elif deployment_mode == "rawdeployment":
+            deployment_mode = "RawDeployment"
+        else:
+            self.unit.status = BlockedStatus(
+                "Please set deployment-mode to either Serverless or RawDeployment"
+            )
+            return
```

Comment:
but this will return None to `cm_resource_handler`

and will proceed code execution. I think it is not what we expected, or?

Reply:
Would it be better to raise ErrorWithStatus here? It will be handled in `_on_event()`

Reply:
looks like yes, the only way

or 1 million returns :D